include/xtensor/xchunked_view.hpp

```diff
@@ -0,0 +1,200 @@
+/***************************************************************************
+* Copyright (c) Johan Mabille, Sylvain Corlay and Wolf Vollprecht          *
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_CHUNKED_VIEW_HPP
+#define XTENSOR_CHUNKED_VIEW_HPP
+
+#include "xstrided_view.hpp"
+#include "xnoalias.hpp"
+
+namespace xt
+{
+
+    template <class E>
+    class xchunked_view;
+
+    template <class E>
+    class xchunk_iterator
+    {
+    public:
+        xchunk_iterator(xchunked_view<E>& chunked_view);
+
+        xchunk_iterator<E>& operator++();
+        xchunk_iterator<E> operator++(int);
+        bool operator==(const xchunk_iterator& other) const;
+        bool operator!=(const xchunk_iterator& other) const;
+        auto operator*();
+
+        friend class xchunked_view<E>;
```

Comment:
This should not be required, the `xchunked_view` is responsible for instantiating iterators only, not for accessing / changing its internal state.